x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/lib/create_timerange.ts

```diff
@@ -14,18 +14,27 @@ export const createTimerange = (
   timeframe?: { end: number; start?: number },
   lastPeriodEnd?: number
 ) => {
-  const to = moment(timeframe ? timeframe.end : Date.now()).valueOf();
+  const end = moment(timeframe && timeframe.end ? timeframe.end : Date.now()).valueOf();
+  const start = moment(timeframe && timeframe.start ? timeframe.start : end).valueOf();
 
   // Rate aggregations need 5 buckets worth of data
   const minimumBuckets = aggType === Aggregators.RATE ? 2 : 1;
-  const calculatedFrom = lastPeriodEnd ? lastPeriodEnd - interval : to - interval * minimumBuckets;
 
-  // Use either the timeframe.start when the start is less then calculatedFrom
-  // OR use the calculatedFrom
+  interval = interval * minimumBuckets;
+
+  let calculatedFrom = end - interval;
+
+  if (lastPeriodEnd && lastPeriodEnd - interval < start) {
+    const maxAllowedLookBack = moment(start).subtract(3 * interval, 'ms');
+    // Calculate the maximum allowable look-back time (3 intervals before the current 'from' time).
+    if (moment(lastPeriodEnd).isAfter(maxAllowedLookBack)) {
+      // Ensure lastPeriodEnd is within the allowable look-back range.
+      calculatedFrom = lastPeriodEnd - interval;
+    }
+  }
   const from =
     timeframe && timeframe.start && timeframe.start <= calculatedFrom
       ? timeframe.start
       : calculatedFrom;
-
-  return { start: from, end: to };
+  return { start: from, end };
```

Comment:
nit: Can we copy the exact code from the observability plugin, or is there a difference? The other one does not need to pass `from` to `start`, and it works with `start` variable.

Reply:
Actually, I wanted to share one file to calculate the time range for both rules. However, as you mentioned, the Custom threshold rule will always have a start and end date, but the Metric one does not. Also, there is a difference in the type of variables. Aligning these would require updating the executor and the related test. So, I kept the PR scope because, as you know, we want to keep our investment low in the Metric rule.